etl/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/etl/phenotype/SequentialLoader.java

```
@@ -141,6 +158,63 @@ private static void loadTemplate() throws FileNotFoundException, IOException {
 		}
 	}
 
+	private static void processRecord(final PhenoCube[] currentConcept, CSVRecord record) {
+		if(record.size()<4) {
+			log.info("Record number " + record.getRecordNumber()
+				+ " had less records than we expected so we are skipping it.");
+			return;
+		}
+
+		try {
+			String conceptPathFromRow = record.get(CONCEPT_PATH);
+			String[] segments = conceptPathFromRow.split("\\\\");
+			for(int x = 0;x<segments.length;x++) {
+				segments[x] = segments[x].trim();
+			}
+			conceptPathFromRow = String.join("\\", segments) + "\\";
+			conceptPathFromRow = conceptPathFromRow.replaceAll("\\ufffd", "");
+			String textValueFromRow = record.get(TEXT_VALUE) == null ? null : record.get(TEXT_VALUE).trim();
+			if(textValueFromRow!=null) {
+				textValueFromRow = textValueFromRow.replaceAll("\\ufffd", "");
+			}
+			String conceptPath = conceptPathFromRow.endsWith("\\" +textValueFromRow+"\\") ? conceptPathFromRow.replaceAll("\\\\[^\\\\]*\\\\$", "\\\\") : conceptPathFromRow;
+			// This is not getDouble because we need to handle null values, not coerce them into 0s
+			String numericValue = record.get(NUMERIC_VALUE);
+			if((numericValue==null || numericValue.isEmpty()) && textValueFromRow!=null) {
+				try {
+					numericValue = Double.parseDouble(textValueFromRow) + "";
+				}catch(NumberFormatException e) {
+
```

Maybe add a comment why we don't want to swallow errors.